src/ansys/dpf/core/server_types.py

```diff
@@ -127,36 +127,32 @@ def _run_launch_server_process(
 def _wait_and_check_server_connection(
     process, port, timeout, lines, current_errors, stderr=None, stdout=None
 ):
-    if not stderr:
-
-        def read_stderr():
-            with io.TextIOWrapper(process.stderr, encoding="utf-8") as log_err:
-                for line in log_err:
-                    LOG.debug(line)
-                    current_errors.append(line)
-
-        stderr = read_stderr
-        # check to see if the service started
-    if not stdout:
-
-        def read_stdout():
-            with io.TextIOWrapper(process.stdout, encoding="utf-8") as log_out:
-                for line in log_out:
-                    LOG.debug(line)
-                    lines.append(line)
-
-        stdout = read_stdout
-    # must be in the background since the process reader is blocking
-    Thread(target=stdout, daemon=True).start()
-    Thread(target=stderr, daemon=True).start()
-
     t_timeout = time.time() + timeout
     started = False
     timedout = False
     while not started and len(current_errors) == 0:
-        # print(lines)
-        started = any("server started" in line for line in lines)
-
+        ## STDOUT parsing
+        # Block for stdout output
+        stdout_line = process.stdout.readline().decode("utf-8")
+
+        # Detect when server is started
+        if stdout_line != None and "server started" in stdout_line:
+            started = True
```

Comment:
@ansys-akarcher shouldn't we also break?

Reply:
Yeah I was wondering about that also

I don't know if it would be possible to have "server started" and some stderr output

in this case the behavior would be different (breaking would wrongly set started as True, skipping any errors)